### DISCUSSION TITLE
CC2531 non rileva dispositivi

### DISCUSSION THREAD

**Discussion by calcioscacchi** [indomus/forum#983] [HUB: Home Assistant]
Ciao a tutti, 

dopo aver flashato il firmware sul CC2531 via raspberry, zigbee2mqtt non riesce a rilevare il dispositivo zigbee che sto cercando di accoppiare (interruttore della vimar 14592).

Non so dire al momento se è il dispositivo o il CC2531. Posso dire però quanto segue:

1) l'interruttore è in modalità zigbee ed è funzionante con alexa, quindi mi viene da dire che lui funziona
2) Dopo qualche minuto, il led verde del CC2531 si spegne, non so se è normale.

Quali log posso consultare per capire dove sta l'errore? al momento, zigbee2mqtt mi da solo questi messaggi di log quando faccio partire la ricerca di dispositivi:

```
Info Zigbee: allowing new devices to join.
Info MQTT publish: topic 'zigbee2mqtt/bridge/response/permit_join', payload '{"data":{"time":254,"value":true},"status":"ok","transaction":"sxavx-1"}'
```

**Comment by sky7176** [comment]
Ciao,
hai provato a disaccoppiare l'interruttore da alexa?
Oppure prova a resettare l'interruttore alle condizioni di default, credo che un dispositivo zigbee possa essere accoppiato ad un solo coordinator.

Giuseppe

**Reply by calcioscacchi** [reply]
Non sapevo di questo limite di massimo una associabilitá.

**Reply by calcioscacchi** [reply]
Avevi ragione! disassociato da alexa, l'interruttore è stato riconosciuto! Purtroppo però ho fatto una amara (ma temuta) scoperta... il rasp che gestisce HA sta chiuso in uno sgabuzzino, e purtroppo l'interruttore non viene raggiunto. Se collego il rasp nella stessa stanza, viene riconosciuto e senza problemi posso accendere e spegnere la luce. Ho ordinato una prolunga USB sperando di migliorare la cosa, altrimenti mi tocca comprare un conbeeII.

Nel frattempo, grazie per la dritta :)

**Reply by sky7176** [reply]
Ciao,
felice di esserti stato di aiuto.
Prendi un ripetirore ZigBee, quelli dell' Ikea vanno nebissimo e fanno anche da carica batterie con uscita USB.
Lo associ al coordinator e poi lo metti in una presa di corrente e per come è pensata la rete ZigBee, ti tipo mesh, ci pensano i dispositivi a creare il persorso per raggiungere il coordinator.

Giuseppe

**Reply by calcioscacchi** [reply]
Ok, appena so di qualcuno chenva all'ikea provvedo. Comunque, provo prima la strada della prolunga, mi sembra incredibile che a distanza di 6 metri la cc2531 non rileva.

**Reply by snakuzzo** [reply]
non è solo questione di distanza...ma ci sono decine di variabili (interferenze wifi, pareti, porte, etc)...poi mettici che la CC2531 è la più "scarsa" di tutte le soluzioni presenti sul mercato...

**Reply by calcioscacchi** [reply]
Se io inserisco in casa vari dispositivi Zigbee (anche vicino alla chiavetta), in teoria il problema si risolve/si minimizza tramite la creazione della rete mesh, i cui dispositivi sono i nodi?

**Reply by snakuzzo** [reply]
I dispositivi che fanno da router sono solitamente quelli alimentati NON a batteria. Gli altri sono semplicemente end device e non contribuiscono all'estensione della rete mesh.
Inoltre non è che sei senza limiti. Dai un'occhiata [QUI](https://www.zigbee2mqtt.io/guide/faq/#i-read-that-zigbee2mqtt-has-a-limit-of-20-devices-when-using-a-cc2530-cc2531-adapter-is-this-true) per il discorso CC2531

**Reply by calcioscacchi** [reply]
dubbio. L'associazione tra i dispositivi avviene tramite l'interfaccia di zigbee2mqtt. come faccio io a specificare che i dispositivi si devono collegare ai dispositivi/router e non direttamente alla CC2531?

**Reply by sky7176** [reply]
Non puoi specificare chi si collega con chi, essendo la rete zigbee di tipo mesh, la topologia della rete viene gestina in automatico dai vari dispositivi cercando il percorso a minor costo e maggior efficenza energetica.

**Reply by calcioscacchi** [reply]
perfetto
